pkgs/applications/misc/stepreduce/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+}:
+
+stdenv.mkDerivation rec {
+  pname = "stepreduce";
+  version = "unstable-2020-04-30";
+
+  src = fetchFromGitLab {
+    owner = "sethhillbrand";
+    repo = "stepreduce";
+    rev = "e89091c33b67e2a18584e1fe3560bfd48ae98773";
+    hash = "sha256-bCseBQ6J3sWFt0kzaRkV11lwzOGvNPebvQ6w4OJaMBs=";
+  };
+
+  installPhase = "install -Dm755 stepreduce $out/bin/stepreduce";
```

Comment:
Please include pre/post hooks:

```
  installPhase = ''
    runHook preInstall

    install -Dm755 stepreduce $out/bin/stepreduce

    runHook postInstall
  '';
```

Reply:
woeps, i let the the style discussion distracted me from this
sorry for the late response, i've included the suggestion, thanks
(it's such an easy thing to miss, is there a better way to do this?)